### PR TITLE
fix(unreal): Stop overwriting a user supplied error level

### LINKED ISSUE #3727: Can't change the error level from an Unreal crash reporter event
Hi!
I'd like to change our crash event from Unreal's crashreporter from "error" to "fatal"
If I'm correct, relay only changes the error level when a crash has the "is asset" or "is ensure" tag. Else it's always using the "error" level
( see https://github.com/getsentry/relay/blob/7b8316f6aecdcc000bc53879ff17338225a5415a/relay-server/src/utils/unreal.rs#L249 ) 
Something like:
```rust
if let Some(explicit_level) = context.engine_data.get("level") {
    event.level = Level::from_str(explicit_level).unwrap();
} else {
    if runtime_props.is_assert.unwrap_or(false) {
        event.level = Annotated::new(Level::Error)
    }
    if runtime_props.is_ensure.unwrap_or(false) {
        event.level = Annotated::new(Level::Warning)
    }
}
```

### PR DESCRIPTION
**Pull request by Dav1dde**
Fixes: #3727

Also updates the test dump with config:
- `IsAssert` is set to `true`
- `__sentry` is extended to contain a `warning` level

### CODE REVIEW

**Review by jjbayer — APPROVED** [review]
LGTM, maybe @bitsandfoxes can confirm that this is the behavior we want.

File: tests/integration/fixtures/native/unreal_crash_with_config
Comment (iker-barriocanal) [inline tests/integration/fixtures/native/unreal_crash_with_config:1]: What does this file look like?

**Review by iker-barriocanal — APPROVED** [review]

**Comment by bitsandfoxes** [comment]
> LGTM, maybe @bitsandfoxes can confirm that this is the behavior we want.

This is exactly the behaviour we're looking for! 🙏

**Review by bitsandfoxes — APPROVED** [review]
Thanks!

**Pull request merged**